sources/packages/api/src/services/institution/institution.service.ts

```diff
@@ -0,0 +1,87 @@
+import { Injectable, Inject } from "@nestjs/common";
+import { RecordDataModelService } from "../../database/data.model.service";
+import { Institution, User } from "../../database/entities";
+import { Connection } from "typeorm";
+import { UserInfo } from "../../types";
+import { CreateInstitutionDto } from "../../route-controllers/institution/models/institution.dto";
+import { LoggerService } from "../../logger/logger.service";
+import { BCeIDService } from "../bceid/bceid.service";
+import { InjectLogger } from "../../common";
+
+@Injectable()
+export class InstitutionService extends RecordDataModelService<Institution> {
+  @InjectLogger()
+  logger: LoggerService;
+  @Inject()
+  bceidService: BCeIDService;
+
+  constructor(@Inject("Connection") connection: Connection) {
+    super(connection.getRepository(Institution));
+    this.logger.log("[Created]");
+  }
+
+  async createInstitution(
+    userInfo: UserInfo,
+    institutionDto: CreateInstitutionDto,
+  ): Promise<Institution> {
+    const institution = this.create();
+    const user = new User();
+    const account = await this.bceidService.getAccountDetails(
+      userInfo.idp_user_name,
+    );
+    console.log("here");
+    console.dir(account);
+    if (account == null) {
+      //TODO: This scenario occurs when basic BCeID users try to push the bceid account into our application.
+      //Ideally this should be handled at the login page where a Basic BCeID user should NOT be allowed to enter the portal
```

Comment:
A ticket was created in the backlog to block BCeIDs. Maybe you could remove the comments if you think that they are not needed anymore.

Reply:
Modified